### PR TITLE
ZAS, GC optimisations, QDel hint fixes

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Oh man this is a doozy.

Ports some ZAS optimisations from Aurora.
~~Also ports ALL OF SSGARBAGE_COLLECTOR FROM AURORA.~~

### CODE REVIEW

File: code/modules/atmospherics/datum_pipeline.dm [hunk 1]
```diff
@@ -15,7 +15,8 @@
 
 /datum/pipeline/Destroy()
 	STOP_PROCESSING(SSprocessing, src)
-	QDEL_NULL(network)
+	if(network)
+		QDEL_NULL(network)
```
Comment (MistakeNot4892) [inline code/modules/atmospherics/datum_pipeline.dm]: Going through this PR to steal lines and noted this - `QDEL_NULL` already falsey-checks the arg.

**Comment by MistakeNot4892** [comment]
`. = ..()` isn't the same as `..()`. `Destroy()` needs to return a valid qdel hint (which is what `. = ..()` is doing) for garbage collection to work properly.

**Comment by BurgerLUA** [comment]
Interesting

**Comment by BurgerLUA** [comment]
@MarinaGryphon Are you okay with this PR?

File: code/modules/atmospherics/datum_pipeline.dm [hunk 2]
```diff
@@ -15,7 +15,8 @@
 
 /datum/pipeline/Destroy()
 	STOP_PROCESSING(SSprocessing, src)
-	QDEL_NULL(network)
+	if(network)
```
Comment (MarinaGryphon) [inline code/modules/atmospherics/datum_pipeline.dm]:
```suggestion
```

File: code/modules/atmospherics/datum_pipeline.dm (hunk 1 above)
Comment (MarinaGryphon) [inline code/modules/atmospherics/datum_pipeline.dm]:
```suggestion
	QDEL_NULL(network)
```

**Pull request merged**